### PR TITLE
Add styleguide

### PR DESCRIPTION
**Pull request by karensawrey**

### CODE REVIEW

**Comment by plaindocs** [comment]
Vale https://docs.errata.ai/vale/scoping#markdown--html

File: styleguide/STYLE.md [hunk 1]
```diff
@@ -0,0 +1,174 @@
+# Styleguide  
+Welcome to the Buildkite styleguide. These are the rules and some useful links for creating new docs.
```
Comment (plaindocs) [inline styleguide/STYLE.md:2]: What do you think about something a little friendlier? 

```suggestion
Welcome to the Buildkite styleguide. These are the guidelines we use to write the docs. 
```

File: styleguide/STYLE.md [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+### Serial commas
+Absolutely yes, yes, and yes.  
+
+### Active/passive voice    
+Generally, use passive voice. It gives a bit more distance from what we're talking about, even if it is a bit longer. There are certainly times for active voice, but it's too nuanced for this guide! [Look here](https://www.aje.com/en/arc/writing-with-active-or-passive-voice/) for some guidance.
+
+### Gender 
+Always use ‘they’, never use ‘he’ or ‘she’. 
+Same, but written by [Microsoft](https://docs.microsoft.com/en-us/style-guide/grammar/nouns-pronouns).
```
Comment (plaindocs) [inline styleguide/STYLE.md:32]: This "Same" confused me a little. How about "More info on writing about X in the Microsoft style guide" or something?

File: styleguide/STYLE.md [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+This section covers the matters that go beyond language and provides guidelines for consistency and a unified look.  
+
+### Consistency
+Keep your writing consistent with itself and other docs. This means abbreviations, capitalization, hyphens, names of UI elements, etc.   
+
+### Title capitalisation  
+
+Use Title case in page headings:
+*"The Quick Brown Fox Jumps Over the Lazy Dog."*  
+Also known as headline style and capital case. All words capitalized, except for certain subsets defined by rules that are not universally standardized, often minor words such as "the" (as above), "of", or "and". The standardization is only at the level of house styles and individual style manuals. (See Headings and publication titles.) A simplified variant is start case, where all words, including articles, prepositions, and conjunctions, start with a capital letter.
```
Comment (plaindocs) [inline styleguide/STYLE.md:44]: We can link "Title.sh as a useful thing here"

File: styleguide/STYLE.md [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+
+Use Sentence case in section headings: 
+*"The quick brown fox jumps over the lazy dog."*  
+The standard case used in English prose. Generally equivalent to the baseline universal standard of formal English orthography mentioned above; that is, only the first word is capitalized, except for proper nouns and other words which are generally capitalized by a more specific rule.
+
+Same, but [on Wikipedia](https://en.wikipedia.org/wiki/Capitalization#By_name_of_style).
```
Comment (plaindocs) [inline styleguide/STYLE.md:50]: Same as above for this same.

File: styleguide/STYLE.md [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+Only capitalize the name of the product - e.g. 'GitHub organization' and not 'GitHub Organization'.  
+
+### Capital letters in UX elements
+Use title capitalization for names of tabs and buttons in the Buildkite interface - e.g. 'Personal Settings', 'Repository Providers', 'Save Organization Settings'.
+
+### Bullet lists  
+
+Capitalize the first word; no full stops at the end if it’s only one sentence. If there are two or more sentences in the list element, the final one will have a full stop.  
+
+Only use bullet lists in the body of the text and never - in page or section headings.
```
Comment (plaindocs) [inline styleguide/STYLE.md:62]: Is this even possible?

File: styleguide/STYLE.md [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+Long numbers use commas to separate thousands - '100,000,000.00'. When in doubt, look [here](https://docs.microsoft.com/en-us/style-guide/numbers).  
+
+Do not use numbers in page headings. Only use numbers less than 10 in section headings.  
+No restrictions on using numbers in the body of the text.  
+
+### Time and date  
+Use 24hr time with hours and minutes, but not seconds. Include timezone. e.g. 17:00 AEST 
+More in [Microsoft Style Guide](https://docs.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/term-collections/date-time-terms).
+
+### Non-breaking characters
```
Comment (plaindocs) [inline styleguide/STYLE.md:78]: Do you think we should remove this section? Or better to leave it in?

File: styleguide/STYLE.md [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+
+Generally not used in Buildkite documentation. If you really feel the need for a non-breaking hyphen, just make sure it’s not in a URL.
+
+### Spacing after full stops  
+**Question:** Should you use one or two spaces after end punctuation?
+**Answer:** One space.
+
+A little [historical background](https://www.onlinegrammar.com.au/the-grammar-factor-spacing-after-end-punctuation-capitals/) on why this is even a valid question.
+!["Two spaces after a period!" vs. "One space after a period" vs. "Line break after every sentence" comic image](https://imgs.xkcd.com/comics/third_way.png "https://xkcd.com/ Third Way comic image")  
+P.S. Remember that, ironically enough, in Markdown, line breaks demand exactly two blank spaces at the end of the line.
```
Comment (plaindocs) [inline styleguide/STYLE.md:89]: WAAAAAAT? No really, WAAAT? TIL

File: styleguide/STYLE.md [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+[Non-breaking spaces](https://www.oxygenxml.com/dita/styleguide/webhelp-feedback/Artefact/Syntax_and_Markup/c_Non_Breaking_Spaces.html) are intended for use when the separation of two adjacent words through line wrapping will result in a loss of meaning or legibility. For example, some authors prefer values and their units, such as "12 kg", to be treated as a single word. Likewise, soft hyphens are intended for use when there is a preferred point for the breaking of a word during line-wrapping. 
+
+Generally not used in Buildkite documentation. If you really feel the need for a non-breaking hyphen, just make sure it’s not in a URL.
+
+### Spacing after full stops  
+**Question:** Should you use one or two spaces after end punctuation?
+**Answer:** One space.
+
+A little [historical background](https://www.onlinegrammar.com.au/the-grammar-factor-spacing-after-end-punctuation-capitals/) on why this is even a valid question.
+!["Two spaces after a period!" vs. "One space after a period" vs. "Line break after every sentence" comic image](https://imgs.xkcd.com/comics/third_way.png "https://xkcd.com/ Third Way comic image")
```
Comment (plaindocs) [inline styleguide/STYLE.md:88]: I love the comic,but I don't think including here is quite right.

File: styleguide/STYLE.md [hunk 9]
```diff
@@ -0,0 +1,174 @@
...
+| buildkite-agent           | When referring to the cli tool, visually should be presented in a code block                     |
+| Sign up/log in            | The action of signing up                                                                         |
+| Signup/login              | When referring to a page that enables signing up or to the signup process                        |
+| API, SSO, SAML            | Always capitalized |
+| GitHub                    | Always capitalized, with an uppercase H in the middle                                            |
+| Two-factor authentication | In a sentence two-factor authentication, in a title Two-Factor Authentication, in short form 2FA |
+| Single sign-on            | In a sentence single sign-on, in a title Single Sign-On, in short form SSO                       |  
+
+### Common trip-ups
+Linter cannot do all of the work for you, so please pay attention to the following cases:
```
Comment (plaindocs) [inline styleguide/STYLE.md:112]:
```suggestion
Linters cannot do all of the work for you, so please pay attention to the following cases:
```

File: styleguide/STYLE.md [hunk 10]
```diff
@@ -0,0 +1,174 @@
...
+
+### Code blocks  
+A code example longer than a couple of words that isn’t part of a sentence/a multi-line code sample needs to be formatted as a code block according to the [GitHub markdown flavor](https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code).
+To add a code block, indent it using four (4) spaces or use 3 backticks (\`\`\`) before and after the code block.
+
+```
+Hello, world!
+```
+This probably goes without saying, but do not use code fragments in page headings or section headings.
+
```
Comment (plaindocs) [inline styleguide/STYLE.md:141]: Let's also mention 

```
{: codeblock-file="pipeline.yml"}
```

immediately after the block to add a filename

Comment (plaindocs) [reply styleguide/STYLE.md:141]: And to add syntax highlighting with Rouge. http://rouge.jneen.net/

    ```bash

according to https://github.com/rouge-ruby/rouge/wiki/List-of-supported-languages-and-lexers

File: styleguide/STYLE.md [hunk 11]
```diff
@@ -0,0 +1,174 @@
...
+app>views>layouts>application.html.erb with a description (e.g. `"G Cloud Identity", 'integrations/sso/g-cloud-identity'` ).
+> **Note:** Ruby, which keeps the website running, interprets underscores in filenames as hyphens. So if a page is called `octopussy_cat.erb.md`, you need to add it as `octopussy-cat` to the `application.html.erb` file.     
+
+
+
+## Screenshots
+This information was aggregated by going over the existing screenshots in the documentation repo. Feel free to change or expand it.
+
+### Taking and processing screenshots  
+**Format:** PNG
```
Comment (plaindocs) [inline styleguide/STYLE.md:154]: We should probably bullets this list so it doesn't look like this;

image

File: styleguide/STYLE.md [hunk 12]
```diff
@@ -0,0 +1,174 @@
...
+
+```
+Hello, world!
+```
+This probably goes without saying, but do not use code fragments in page headings or section headings.
+
+### Adding and naming new documentation pages  
+To add a new documentation page, create it as a *.md.erb file. Give it a lowercase name, separate words using underscores.
+To add the new page to the documentation sidebar on https://buildkite.com/docs, add the corresponding entry to 
+app>views>layouts>application.html.erb with a description (e.g. `"G Cloud Identity", 'integrations/sso/g-cloud-identity'` ).
```
Comment (plaindocs) [inline styleguide/STYLE.md:145]: What about

```suggestion
`app/views/layouts/application.html.erb` with a description (e.g. `"G Cloud Identity", 'integrations/sso/g-cloud-identity'` ).
```

**Review by plaindocs — COMMENTED** [review]
This is looking great! I've made a few suggestions here and there.

**Comment by plaindocs** [comment]
We can also  mention Docs Note and Docs Troubleshooting Note:

```
<section class="Docs__note">
  <h3>Setting agent defaults</h3>
  <p>Use a top-level <code>agents</code> block to <a href="/docs/pipelines/defining-steps#step-defaults">set defaults</a> for all steps in a pipeline.</p>
</section>

<section class="Docs__troubleshooting-note">
  <p class="Docs__note__heading">Running each build in it’s own container</p>
  <p>This page references the out-of-date Buildkite Agent v2.</p>
  <p>For docs referencing the Buildkite Agent v3, <a href="/docs/agent/v3/cli_artifact">see the latest version of this document</a>.
</section>
```

We should standardize on title for these I think.

File: styleguide/STYLE.md (hunk 7 above)
Comment (karensawrey) [reply styleguide/STYLE.md:89]: You live - you learn ;)

File: styleguide/STYLE.md (hunk 6 above)
Comment (karensawrey) [reply styleguide/STYLE.md:78]: Killed the beast.

File: styleguide/STYLE.md [hunk 13]
```diff
@@ -0,0 +1,224 @@
...
+* [Language](#Language)
+* [Style and formatting](#Style-and-formatting)  
+* [Code and filenames](#Code-and-filenames)
+* [Screenshots](#Screenshots)
+
+## Language  
+This section covers everything related to the language and formatting used in Buildkite documentation (and website, to some extent).
+
+### Dictionary  
+We use American English, so our first-call dictionary is [Merriam Webster](https://www.merriam-webster.com/ ). It is not a single source of truth, just a good starting/reference point.
```
Comment (plaindocs) [inline styleguide/STYLE.md:15]:
```suggestion
We use American English, so our first-call dictionary is [Merriam Webster](https://www.merriam-webster.com/). It is not a single source of truth, just a good starting/reference point.  
```

File: styleguide/STYLE.md [hunk 14]
```diff
@@ -0,0 +1,224 @@
...
+
+## Language  
+This section covers everything related to the language and formatting used in Buildkite documentation (and website, to some extent).
+
+### Dictionary  
+We use American English, so our first-call dictionary is [Merriam Webster](https://www.merriam-webster.com/ ). It is not a single source of truth, just a good starting/reference point.  
+We also use [Vale](https://github.com/apps/vale-linter) and our own [linter](https://github.com/buildkite/docs#linting) with a file that contains our own [custom list of words](https://github.com/buildkite/docs/blob/master/vale/vocab.txt) that may confuse the spellchecker.
+
+### English flavor 
+Again, we use American English. We're also a highly multi-national team, so here is the list of the most notable [differences between American, British, and Australian English](http://linktranslation.com.au/the-differences-between-american-british-and-australian-english/) Eglish to watch out for.
```
Comment (plaindocs) [inline styleguide/STYLE.md:19]:
```suggestion
Again, we use American English. We're also a highly multi-national team, so here is the list of the most notable [differences between American, British, and Australian English](http://linktranslation.com.au/the-differences-between-american-british-and-australian-english/) English to watch out for.  
```

File: styleguide/STYLE.md [hunk 15]
```diff
@@ -0,0 +1,224 @@
...
+
+### Headings  
+No multi-sentence headings or full stops at the end of a sentence in page or sentence headings. 
+Be consistent about heading levels - H1, H2, H3 - no jumping from H1 to H3 or H2 to H4.
+
+## Code and filenames  
+This section deals with adding and properly formatting code in the documentation + naming files, pages, and their derivative URLs.  
+
+### Code formatting  
+We use the basic GitHub flavor of markdown for [formatting code](https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code).
```
Comment (plaindocs) [inline styleguide/STYLE.md:117]:
```suggestion
We use the GitHub flavor of markdown for [formatting code](https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code). 
```

File: styleguide/STYLE.md [hunk 16]
```diff
@@ -0,0 +1,224 @@
...
+```
+<!-- vale off -->
+
+This is some text
+
+more text here...
+
+<!-- vale on -->
+```
+Use the `vale on` syntax before a phrase that needs to be bypassed by the linter and don't forget to turn it on again with `vale on`.
```
Comment (plaindocs) [inline styleguide/STYLE.md:169]:
```suggestion
Use the `vale off` syntax before a phrase that needs to be bypassed by the linter and don't forget to turn it on again with `vale on`.  
```

File: styleguide/STYLE.md [hunk 17]
```diff
@@ -0,0 +1,224 @@
...
+### Escaping vale linting  
+
+If you absolutely need to add some word that triggers the linter, you can use escaping via the following syntax: 
+
+```
+<!-- vale off -->
+
+This is some text
+
+more text here...
```
Comment (plaindocs) [inline styleguide/STYLE.md:165]:
```suggestion
This is some text that you do NOT want the linter to check
```

File: styleguide/STYLE.md [hunk 18]
```diff
@@ -0,0 +1,224 @@
...
+
+<!-- vale on -->
+```
+Use the `vale on` syntax before a phrase that needs to be bypassed by the linter and don't forget to turn it on again with `vale on`.  
+
+### Custom elements  
+We have a few custom scripts for adding useful custom elements that are missing in vanilla Markdown.
```
Comment (plaindocs) [inline styleguide/STYLE.md:172]:
```suggestion
We have a few custom scripts for adding useful elements that are missing in Markdown.  
```

**Review by plaindocs — APPROVED** [review]
Super good! I've left a couple of quibbles inline, lets fix those and merge this.

@harrietgrace take a look if you fancy

File: styleguide/STYLE.md [hunk 19]
```diff
@@ -0,0 +1,224 @@
...
+| We (as in Buildkite) | No                                                        | Yes                                                                     | Yes                                                                     | Avoid if possible; use 'I' if you need to                                 |
+| Links                | Relative paths to other docs, full paths to anything else | Always full paths, check for HTTPS and that you’re not using .localhost | Always full paths, check for HTTPS and that you’re not using .localhost | Always full paths, check for HTTPS and that you’re not using .localhost |
+| Exclamation marks    | No                                                        | Yes, although use with restraint, and not more than one in a row        | Yes, although use with restraint, and not more than one in a row        | Yes, although use with restraint, and not more than one in a row        |
```
Comment (plaindocs) [inline styleguide/STYLE.md:87]: This should  display better as well

```suggestion
|                      | Docs                                                      | Twitter and Blog                                                        | Changelog                                                               |
|----------------------|-----------------------------------------------------------|-------------------------------------------------------------------------|-------------------------------------------------------------------------|
| We (as in Buildkite) | No                                                        | Yes                                                                     | Avoid if possible; use 'I' if you need to                               |
| Links                | Relative paths to other docs, full paths to anything else | Always full paths, check for HTTPS and that you’re not using .localhost | Always full paths, check for HTTPS and that you’re not using .localhost |
| Exclamation marks    | No                                                        | Yes, although use with restraint, and not more than one in a row        | Yes, although use with restraint, and not more than one in a row        |
```

**Comment by karensawrey** [comment]
All clear, ready for merge.

File: styleguide/STYLE.md [hunk 20]
```diff
@@ -0,0 +1,224 @@
...
+To add a code block, indent it using four (4) spaces or use 3 backticks (\`\`\`) before and after the code block.
+
+```
+Hello, world!
+```
+
+To add a filename to a codeblock, immediately after the block use `{: codeblock-file="filename.extension"}` (for example, `{: codeblock-file="pipeline.yml"}`).   
+
+To add syntax highlighting, you can use [Rouge](http://rouge.jneen.net/), for example:  
+```
```
Comment (plaindocs) [inline styleguide/STYLE.md:137]: It has just occurred to that these code blocks probably don't nest well.

File: styleguide/STYLE.md [hunk 21]
```diff
@@ -0,0 +1,224 @@
...
+```bash
+#!/bin/sh
+echo "Hello world"```
+```  
+turns into 
+```bash
+#!/bin/sh
+echo "Hello world"
+```
```
Comment (plaindocs) [inline styleguide/STYLE.md:146]:
```suggestion
    ```bash
    #!/bin/sh
    echo "Hello world"
    ```  
```

this is a little simpler, and should do the trick

Comment (plaindocs) [reply styleguide/STYLE.md:146]: Up to you, and apologies for reviving a completed PR. ;-)

**Comment by plaindocs** [comment]
:tada:

**Pull request merged**